pytaps/transports.py

```diff
@@ -121,7 +121,7 @@ class UdpTransport(TransportLayer):
 
     async def close(self):
         print_time("Closing connection.", color)
-        self.transport.close()
+        # self.transport.close()
```

Comment:
Why is this commented out now?

Reply:
I commented that because I hit an error and wasn't sure how to handle it, and then forgot.  The error looked like this, and went away after removing that line:

~~~
2019-11-02 17:15:02.077919: Received 125-byte datagram
received packet len 125
2019-11-02 17:15:03.103180: Received 125-byte datagram
received packet len 125
2019-11-02 17:15:04.095427: Received 125-byte datagram
2019-11-02 17:15:04.489903: leaving multicast session.
2019-11-02 17:15:04.490542: Closing connection.
Task exception was never retrieved
future: <Task finished coro=<UdpTransport.close() done, defined at /home/jholland/venv-taps/lib/python3.7/site-packages/pytaps-0.1-py3.7-linux-x86_64.egg/pytaps/transports.py:122> exception=AttributeError("'UdpTransport' object has no attribute 'transport'")>
Traceback (most recent call last):
  File "/home/jholland/venv-taps/lib/python3.7/site-packages/pytaps-0.1-py3.7-linux-x86_64.egg/pytaps/transports.py", line 124, in close
    self.transport.close()
AttributeError: 'UdpTransport' object has no attribute 'transport' 
~~~

I *think* this was just a mistake, and that the transport object intentionally doesn't have a transport member, and that the line should be removed instead of commented out?

Reply:
The "transport" that gets closed there is different from UdpTransport object. I will rename it in the future to make it more clear. (Sadly asyncio has a class called transport that is doing something other than our TransportLayer classes. 
The problem here is that if the UDP connection gets initiated in the normal unicast way a aiotransport object gets created by the asyncio createDatagramEndpoint call (or more specifically in the connectionMade callback from that call). Since we neve have an aiotransport object when using multicast it is missing here when its supposed to be closed. The easiest solution would be to check if we have unicast and only close the aiotransport in that case. I assume all the cleanup that is required when closing the multicast connection is done in your multicast library anyway.

Reply:
OK, I've re-added the transport.close call.  Interestingly, it no longer fails this morning.  I'm not convinced this is correct as it stands, but I'm convinced I probably shouldn't have touched it as part of this set of edits, and that it should be treated as a separate issue if I can reproduce it.  Thanks for clarifying.